triton/tut/parallel.rst

```diff
@@ -3,54 +3,78 @@ Parallel computing
 ==================
 
 Parallel computing is what HPC is really all about: processing things on
-more than one CPU at once. By now, you should have read all of the previous
+more than one processor at once. By now, you should have read all of the previous
 tutorials.
 
 Parallel programming models
 ---------------------------
 
-Parallel programming is a completely different way of programming.  Most
-Triton users don't need to write their own
-applications, at most they will be running existing programs, but in
-order to understand things, we start with some introduction.
+Parallel programming is used to create programs that can execute
+instructions on multiple processors at a same time. Most of our users that
+run their programs in parallel utilize existing parallel execution features
+that are present in their programs and thus do not need to learn how to create
+parallel programs. But even when one is running programs in parallel,
+it is important to understand different models of parallel execution.
 
 The two main models are:
 
-* Shared memory program (or multithreading) runs on only one node
-  because, like the name says, all the memory has to be accessible to
-  all the processes.  Thus, scaleability is limited to a number of CPU
-  cores available within one computational node. The code is
-  easier to implement and the same code can still be run in a serial mode.
-  Examples of application that utilize this model: Matlab, R, OpenMP
-  applications, typical parallel desktop programs.
+* Shared memory (or multithreaded) programs run multiple independent
+  workers on the same machine. As the name suggests, all of the computer's
+  memory has to be accessible to all of the processes. **Thus programs
+  that utilize this model are limited to one node only.** Likewise, the
+  maximum number of workers is usually the number of CPU cores available
+  on the computational node. The code is easier to implement and the same
+  code can still be run in a serial mode. Example applications that
+  utilize this model: Matlab, R, Python multithreading/multiprocessing,
+  OpenMP applications, BLAS libraries, FFTW libraries, typical
+  multithreaded parallel pesktop programs.
 
 * Message passing programming (e.g. MPI, message passing interface)
   can run on multiple nodes interconnected with the network via passing
-  data through MPI software libraries. The large-scale scientific programs
-  are MPI. MPI can scale to thousands of CPU cores, but it's harder to
-  implement from the programmer point of view.
+  data through MPI software libraries. Almost all large-scale scientific
+  programs utilize MPI. MPI can scale to thousands of CPU cores, but
+  depending on the case it can be harder to implement from the
+  programmer's point of view. Example applications that utilize this
+  model: CP2K, GPAW, LAMMPS, OpenFoam.
```

Comment:
```suggestion
  model: CP2K, GPAW, LAMMPS, OpenFoam.  **For MPI programs, you
  request one/multiple nodes and multiple processors, and usually start one
  process for each processor - and they figure out how to communicate themselves.**
```

Reply:
I added this information in another example.